### DISCUSSION TITLE
Android Support

### DISCUSSION THREAD

**Discussion by TheKing349** [Voltstro-Studios/UnityWebBrowser#131] [Q&A] [answered]
I know you've said mobile support won't work, but could you explain how it wouldn't work? I'm trying to get something for Oculus Quest 2, and your project looked intriguing. I just want to know how/why the project wouldn't work with Android, if I just switch the target platform to be Android. Thanks!

**Comment by Voltstro** [comment, marked as answer]
I don't plan on supporting mobile devices for a few reasons:
- UWB relies on the usage of an "engine", of which is a background process that is actually responsible for doing the web engine things. Now Android does support running background processes (Android calls them services), but now you gotta consider the additional overhead of a completely separate background service on a mobile devices.
- CEF, which is used by UWB's only current engine, doesn't support android (or any Linux-ARM devices), which leads into my third reason:
- Android has `WebView`, you would be better off using that directly, rather then having an external engine talk to `WebView` than pipe it back, as `WebView` doesn't have the issues that running external engine processes are solving.

**Reply by TheKing349** [reply]
Okay that makes sense! Thank you for your time, as well as your answer! Your project looks amazing though, may have to play around with it later on!

**Reply by kris-w-dev** [reply]
hi , If we use Chromium instead of cef, can we support Android?

**Reply by kris-w-dev** [reply]
I hope my website can run on VR devices. This website needs to use VideoEncoder, but webview and gecko do not support VideoEncoder, So I want to know why Chromium doesn't support Android?

**Reply by TheKing349** [reply]
It may be possible, but who knows. It may require a different way/setup for chromium to work

**Reply by Voltstro** [reply]
> hi , If we use Chromium instead of cef, can we support Android?

I don't understand what you mean. The Chromium project itself is built and designed as an application, there is no other way of using it as a library without the use of a library wrapper (CEF).
